waiter/src/waiter/service_description.clj

```diff
@@ -996,7 +1008,11 @@
   [attach-service-defaults-fn attach-token-defaults-fn token-sequence token->token-data]
   (let [merged-token-data (attach-token-defaults-fn
                             (token-sequence->merged-data token->token-data token-sequence))
-        service-description-template (select-keys merged-token-data service-parameter-keys)
+        exclusive-mode? (= "exclusive" (get merged-token-data "service-mapping"))
+        service-description-template (cond-> (select-keys merged-token-data service-parameter-keys)
+                                       ;; each unique token permutation will create a unique service
+                                       ;; leverage WAITER_CONFIG_ prefixed environment variables being allowed
```

Comment:
I don't particularly like this hack (plumbing the token through in this quasi-reserved environment variable, and then dropping it later) — but I'm not sure it's worth the effort right now to figure out another way to propagate the token value. Since the `WAITER_CONFIG_` values should be considered quasi-reserved anyway, I don't think it would be a problem if we came back and decided to clean this up later (i.e., it wouldn't break any contracts that users should be depending on) for the same reason that reserving it for internal use in this PR is OK.

Reply:
An alternate is to replace the `service-mapping` value with the token value. Would you prefer that approach?

Reply:
Also, I can avoid dropping it and sticking to the `WAITER_CONFIG_TOKEN` name, but I prefer we use the `WAITER_TOKEN` environment variable.

Reply:
I also prefer `WAITER_TOKEN`. Let's just leave it as-is.